split The New Architecture guide into few pages

Simek opened the pull request:
This PR splits the New Architecture guide into few smaller pages which all are located in the "The New Architecture" section in the sidebar. I hope this will improve the overall readability of the guide for the users, since the split has been request few times.

The new pages titles can be improved for sure, I was focusing on trying to fit them in one line, but maybe it is an overoptimization, hope someone will look at that later. 😅 

In this PR I have been focusing on the layout and formatting, no on the content itself, however there were small whitespace changes and syntax highlight correction, applied while I was moving the content.

To achieve the effect visible below I had to alter few lines of CSS, because non-linkable sidebar entries were missing a proper style (they were not used before) and we were unnecessary overwriting the sidebar sub-sections font size due to class name changes on the Docusuaurs side. Those changes will be pushed in separate PR against `main` branch.

### Preview

<img width="1408" alt="Screenshot 2021-12-16 at 13 03 53" src="https://user-images.githubusercontent.com/719641/146373627-16efef19-c2b5-4919-ac10-93fbcbd458da.png">

cortinico reviewed (approved):
The change looks great to me 👍 Thanks for taking the time to do this @Simek and sorry for the late review.

Are you fine with me pushing to your branch to fix the merge conficts? Or do you wish to do it by yourself? I've pushed several fixes on the playbook PR that should be included in the page split as well.

Simek commented:
I have resolved the conflicts and updated the branch with `feat/new-arch` and then synced it with `main` to get access to new "Architecture" top level grouping, to which I have moved the playbook entries.

### Updated preview

<img width="1429" alt="Screenshot 2022-01-01 at 17 16 26" src="https://user-images.githubusercontent.com/719641/147854978-7f0174bf-d0d4-40a9-b272-3fe60a01e112.png">

cortinico commented:
> I have reverted the `main` synce becouse it would result in issues while merging the `feat/new-arch` target branch into `main`. Let's sort the placement in separate PR, after the merge of this one and sync with `main` in main playbook draft. 😉

Makes total sense 👍 Let's merge this and rebase #2879 on top of `main` afterwards.
